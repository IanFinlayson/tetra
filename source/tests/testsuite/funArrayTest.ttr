def main():
   passer = [1,2,3]
   print("Just created passer and passer = ", passer, "\n")
   cally(passer)
   print("In main, just returned from cally(passer) and passer = ",passer, "\n")
   print("passer[0] + passer[1] + passer[2] = ", passer[0] + passer[1] + passer[2], "\n")

def cally(argx [int]):
   print("Just entered cally() and argx = ", argx, "\n")
   caller(argx)
   print("In cally(), just returned from caller(argx), and argx = ", argx, "\n")

def caller(argy [int]):
   print("Just entered caller() and argy = ", argy, "\n")
   argy[0] = 10
   print("Just set argy[0] = 10 and argy = ", argy, "\n")
   argy[1] = 20
   print("Just set argy[1] = 20 and argy = ", argy, "\n")
   argy[2] = 30
   print("Just set argy[2] = 30 and argy = ", argy, "\n")
