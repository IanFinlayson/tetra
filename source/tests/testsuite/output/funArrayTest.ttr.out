Just created passer and passer = [1, 2, 3]
Just entered cally() and argx = [1, 2, 3]
Just entered caller() and argy = [1, 2, 3]
Just set argy[0] = 10 and argy = [10, 2, 3]
Just set argy[1] = 20 and argy = [10, 20, 3]
Just set argy[2] = 30 and argy = [10, 20, 30]
In cally(), just returned from caller(argx), and argx = [10, 20, 30]
In main, just returned from cally(passer) and passer = [10, 20, 30]
passer[0] + passer[1] + passer[2] = 60

